impress-org/givewp
# Fix: wrong donation amount sent to PayPal if donor changes amount

Resolves [GIVE-309]

## Description



The wrong amount is sent to PayPal if the donor changes the donation amount after filling in the credit card fields in PayPal's debit/credit smart button.

This PR fixes this problem and adds a new verification to check if the amount was changed. If yes, it updates the PayPal amount order before approving it as it's not possible to update the order after it gets approved - status set as COMPLETE.

**Note for reviewer:** The IDE auto-format applied our code style in the SmartButtons.js file that was using another code style, which makes it hard to review. So, to better understand, you can check for the `@unreleased` tags to see what was changed.

## Affects



PayPal Donations

## Visuals



https://www.loom.com/share/da88c326f0414da889f0f2a6fec013af?sid=662ab813-00f2-41cb-a416-7a57697267e2

## Testing Instructions



You can follow these steps for both V2 and V3 forms:

1. Setup PayPal Donations with the Stmart Buttons enabled;
2. On the checkout page select the black button to finish the donation with a debit or credit card;
3. After filling the credit card information, change the donation amount;
4. Complete the donation and check in the PayPal dashboard if the new amount was applied.

## Pre-review Checklist






[GIVE-309]: https://stellarwp.atlassian.net/browse/GIVE-309?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
